ckanext/ytp/request/logic/action/get.py

```diff
@@ -30,7 +30,7 @@ def member_request(context, data_dict):
     member_dict['organization_name'] = membership.group.name
     member_dict['group_id'] = membership.group_id
     member_dict['role'] = member_request.role
-    member_dict['state'] = 'pending'
+    member_dict['state'] = member_request.status
```

Comment:
This seems really wierd to me.  It seems like such an obvious fault on the part of the original coders that I almost wonder if we are missing something. Anyhow time will tell...

Reply:
It the old logic this was the only state the member_request would be in. Otherwise the code was raising a logic exception NotFound.
The other place the issue could be fixed could be:
`ckanext-ytp-request/ckanext/ytp/request/controller.py:72`
```
    def show(self, mrequest_id):
        """"
        Shows a single member request. To be used by admins in case they
        want to modify granted role or accept via e-mail.
        """
        context = {'user': c.user or c.author}
        try:
            membershipdto = toolkit.get_action('member_request_show')(
                context, {'mrequest_id': mrequest_id})
            member_user = model.Session.query(
                model.User).get(membershipdto['user_id'])
            context = {'user': member_user.name}
            roles = self._get_available_roles(
                context, membershipdto['organization_name'])
            extra_vars = {"membership": membershipdto,
                          "member_user": member_user, "roles": roles}
            return render('request/show.html', extra_vars=extra_vars)
        except logic.NotFound:
            abort(404, self.request_not_found_message)
        except logic.NotAuthorized:
            abort(401, self.not_auth_message)
````
Where we could just render a different template to display message "Member request already accepted or not found".